src/instrumentsscene/view/stafftreeitem.cpp

```diff
@@ -29,11 +29,14 @@ using namespace mu::notation;
 StaffTreeItem::StaffTreeItem(IMasterNotationPtr masterNotation, INotationPtr notation, QObject* parent)
     : AbstractInstrumentsPanelTreeItem(InstrumentsTreeItemType::ItemType::STAFF, masterNotation, notation, parent)
 {
-    connect(this, &AbstractInstrumentsPanelTreeItem::isVisibleChanged, [this](bool isVisible) {
+    connect(this, &AbstractInstrumentsPanelTreeItem::isVisibleChanged, [this](bool isVisible)
+    {
         if (!m_isInited) {
             return;
         }
-
+        if (isVisible && !this->parentItem()->isVisible()) {
+            this->parentItem()->setIsVisible(true, false);
+        }
         this->notation()->parts()->setStaffVisible(id(), isVisible);
     });
```

Comment:
It would be great to add some newlines here for readability:
```suggestion

        if (isVisible && !this->parentItem()->isVisible()) {
            this->parentItem()->setIsVisible(true, false);
        }

        this->notation()->parts()->setStaffVisible(id(), isVisible);
    });
```

Reply:
Hey, thank you for the suggestion. Couldn't get back before due to some college work :( I've made the changes :D